### ISSUE TITLE
Fatal error: Index out of range.

### ISSUE THREAD

**Issue by justbcuz** [vermont42/GpxLocationManager#26]
There is a fatal crash with index out of range at line 167 in GpxLocationManager.swift. 

Look like the ">=" of the if statement on line 166 is incorrect.

![screen shot 2018-11-07 at 9 53 10 am](https://user-images.githubusercontent.com/186479/48143388-4a928480-e274-11e8-9b68-47869b4a514d.png)

![screen shot 2018-11-07 at 9 53 41 am](https://user-images.githubusercontent.com/186479/48143393-4e260b80-e274-11e8-99c6-699abeb545f6.png)

**Comment by nkanetka** [comment]
Yikes!!! Thanks for catching this. There are a few things that I'd like to see changed here (safe grabbing of collections 🤔). We'll get a fix for this soon :)

**Comment by vermont42** [comment]
@justbcuz Are you able to share your test data? Also, may I ask how you might be planning to use `GpxLocationManager`? Knowing that the framework is being used in real apps is excellent motivation for Nehal and me to maintain it.

I detect an [Astérix](https://www.adforum.com/creative-work/ad/player/34496235/mustache-2/parc-asterix) [mustache](https://www.grammarly.com/blog/mustache-moustache/) in your avatar photo.

**Comment by justbcuz** [comment]
@vermont42 Yes, there maybe some sort of Astérisk shenanigans going on there :)

Please find below a ZIP if the gpx file I am testing with.
[Luv-hq-30mph.gpx.zip](https://github.com/vermont42/GpxLocationManager/files/2562697/Luv-hq-30mph.gpx.zip)

I am writing a persistent data POC to test realtime data synchronization and was using location updates as the trigger to update a stored object. To see the data synchronization across devices (and not leave my desk) I wanted to test locally using multiple simulators, but Simulator only attaches the Debug->Location->Freeway Drive to the active Xcode debug session. Therefor I could see the data sync going oneway.

This is when I found GpxLocationManager. I love the passthrough wrapper around CLLocationManager, it makes local Location simulation testing an easy add.

`#if targetEnvironment(simulator)`
`let _locationManager: LocationManager = LocationManager(type: .gpxFile("Luv-hq-30mph"))`
`#else`
`let _locationManager: LocationManager = LocationManager(type: .coreLocation)`
`#endif`

I plan on using this more and more.

**Comment by nkanetka** [comment]
#27 Helps stop index out of bound errors from occurring in the future :)
